### ISSUE TITLE
W&B mAP crashes on resume 

### ISSUE THREAD

**Issue by wartek69** [ultralytics/yolov5#6844]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

Hey
I'm using colab to train my model and use the W&B resume functionality to continue training after a timeout.
I've made an issue before that there was an error when I use resume, you suggested to downgrade to version 0.12.10 so I did.
Yesterday I used the resume and it worked properly on that version. However, today when I resumed the training of yesterday my mAP crashed to values 0.00591 and kept growing from there instead of continuing from the mAP 0.4645 that was obtained yesterday.
It looks like the weights of the previous model were not saved and we just started training from scratch.

<details>
<summary>Log Output</summary>
<br>

```
Downloading https://ultralytics.com/assets/Arial.ttf to /root/.config/Ultralytics/Arial.ttf...
wandb: Currently logged in as: alex69 (use `wandb login --relogin` to force relogin)
train: weights=yolov5s.pt, cfg=, data=data/coco128.yaml, hyp=data/hyps/hyp.scratch-low.yaml, epochs=300, batch_size=16, imgsz=640, rect=False, resume=wandb-artifact://alex69/training_logs/18xq5ecx, nosave=False, noval=False, noautoanchor=False, evolve=None, bucket=, cache=None, image_weights=False, device=, multi_scale=False, single_cls=False, optimizer=SGD, sync_bn=False, workers=8, project=runs/train, name=exp, exist_ok=False, quad=False, cos_lr=False, label_smoothing=0.0, patience=100, freeze=[0], save_period=5, local_rank=-1, entity=None, upload_dataset=False, bbox_interval=-1, artifact_alias=latest
github: up to date with https://github.com/ultralytics/yolov5 ✅
YOLOv5 🚀 v6.1-11-g63ddb6f torch 1.10.0+cu111 CUDA:0 (Tesla T4, 15110MiB)

hyperparameters: lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.05, cls=0.5, cls_pw=1.0, obj=1.0, obj_pw=1.0, iou_t=0.2, anchor_t=4.0, fl_gamma=0.0, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0
TensorBoard: Start with 'tensorboard --logdir runs/train', view at http://localhost:6006/
wandb: wandb version 0.12.11 is available!  To upgrade, please run:
wandb:  $ pip install wandb --upgrade
wandb: Tracking run with wandb version 0.12.10
wandb: Resuming run sweet-glitter-7
wandb: ⭐️ View project at https://wandb.ai/alex69/training_logs
wandb: 🚀 View run at https://wandb.ai/alex69/training_logs/runs/18xq5ecx
wandb: Run data is saved locally in /content/yolov5/wandb/run-20220303_092703-18xq5ecx
wandb: Run `wandb offline` to turn off syncing.

wandb: Downloading large artifact run_18xq5ecx_model:latest, 54.90MB. 1 files... Done. 0:0:0
wandb: Downloading large artifact train:latest, 1168.48MB. 3282 files... Done. 0:0:0
wandb: Downloading large artifact val:latest, 257.40MB. 822 files... Done. 0:0:0
wandb: Downloading large artifact val:latest, 257.40MB. 822 files... Done. 0:0:0
Mapping dataset
100% 410/410 [00:00<00:00, 1409561.18it/s]

                 from  n    params  module                                  arguments                     
  0                -1  1      3520  models.common.Conv                      [3, 32, 6, 2, 2]              
  1                -1  1     18560  models.common.Conv                      [32, 64, 3, 2]                
  2                -1  1     18816  models.common.C3                        [64, 64, 1]                   
  3                -1  1     73984  models.common.Conv                      [64, 128, 3, 2]               
  4                -1  2    115712  models.common.C3                        [128, 128, 2]                 
  5                -1  1    295424  models.common.Conv                      [128, 256, 3, 2]              
  6                -1  3    625152  models.common.C3                        [256, 256, 3]                 
  7                -1  1   1180672  models.common.Conv                      [256, 512, 3, 2]              
  8                -1  1   1182720  models.common.C3                        [512, 512, 1]                 
  9                -1  1    656896  models.common.SPPF                      [512, 512, 5]                 
 10                -1  1    131584  models.common.Conv                      [512, 256, 1, 1]              
 11                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']          
 12           [-1, 6]  1         0  models.common.Concat                    [1]                           
 13                -1  1    361984  models.common.C3                        [512, 256, 1, False]          
 14                -1  1     33024  models.common.Conv                      [256, 128, 1, 1]              
 15                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']          
 16           [-1, 4]  1         0  models.common.Concat                    [1]                           
 17                -1  1     90880  models.common.C3                        [256, 128, 1, False]          
 18                -1  1    147712  models.common.Conv                      [128, 128, 3, 2]              
 19          [-1, 14]  1         0  models.common.Concat                    [1]                           
 20                -1  1    296448  models.common.C3                        [256, 256, 1, False]          
 21                -1  1    590336  models.common.Conv                      [256, 256, 3, 2]              
 22          [-1, 10]  1         0  models.common.Concat                    [1]                           
 23                -1  1   1182720  models.common.C3                        [512, 512, 1, False]          
 24      [17, 20, 23]  1     29667  models.yolo.Detect                      [6, [[10, 13, 16, 30, 33, 23], [30, 61, 62, 45, 59, 119], [116, 90, 156, 198, 373, 326]], [128, 256, 512]]
Model Summary: 270 layers, 7035811 parameters, 7035811 gradients, 15.9 GFLOPs

Transferred 349/349 items from artifacts/run_18xq5ecx_model:v6/last.pt
Scaled weight_decay = 0.0006542205810546875
optimizer: SGD with parameter groups 57 weight (no decay), 60 weight, 60 bias
albumentations: version 1.0.3 required by YOLOv5, but version 0.1.12 is currently installed
train: Scanning 'artifacts/train:v0/data/labels' images and labels...1640 found, 0 missing, 418 empty, 0 corrupt: 100% 1640/1640 [00:05<00:00, 288.60it/s]
train: New cache created: artifacts/train:v0/data/labels.cache
val: Scanning 'artifacts/val:v1/data/labels' images and labels...410 found, 0 missing, 82 empty, 0 corrupt: 100% 410/410 [00:01<00:00, 319.26it/s]
val: New cache created: artifacts/val:v1/data/labels.cache
Image sizes 1280 train, 1280 val
Using 2 dataloader workers
Logging results to runs/train/exp
Starting training for 300 epochs...

     Epoch   gpu_mem       box       obj       cls    labels  img_size
    70/299     11.5G   0.05022    0.3252 9.618e-05        13      1280: 100% 118/118 [03:10<00:00,  1.61s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 15/15 [00:23<00:00,  1.54s/it]
                 all        410        762      0.694     0.0214    0.00591    0.00173

     Epoch   gpu_mem       box       obj       cls    labels  img_size
    71/299       11G   0.07144    0.3117 0.0002217         4      1280: 100% 118/118 [03:02<00:00,  1.54s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 15/15 [00:21<00:00,  1.46s/it]
                 all        410        762      0.249      0.117     0.0571     0.0177

     Epoch   gpu_mem       box       obj       cls    labels  img_size
    72/299       11G   0.06371    0.2931 0.0002075         1      1280: 100% 118/118 [03:03<00:00,  1.56s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 15/15 [00:20<00:00,  1.36s/it]
                 all        410        762      0.253     0.0808       0.06     0.0197

     Epoch   gpu_mem       box       obj       cls    labels  img_size
    73/299       11G   0.06032     0.283 0.0001971         7      1280: 100% 118/118 [03:02<00:00,  1.55s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 15/15 [00:21<00:00,  1.40s/it]
                 all        410        762       0.34      0.153      0.118     0.0445

     Epoch   gpu_mem       box       obj       cls    labels  img_size
    74/299       11G   0.05654    0.2659 0.0001982        10      1280: 100% 118/118 [02:58<00:00,  1.52s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 15/15 [00:21<00:00,  1.40s/it]
                 all        410        762      0.333      0.189      0.138     0.0515

```
</details>

You can see the mAP drop in the following screenshot.
![W B Chart 3_3_2022, 10_41_32 AM](https://user-images.githubusercontent.com/26298408/156545966-66a24798-53e4-423b-bb31-f45a73113536.png)
Any clue why this might be happening and how to solve it?

Kind regards
Alex

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by AyushExel** [comment]
@wartek69 there can be many reasons for this unfortunately there's no such thing as a perfect resume. We just try to remember all parameters and models and attempt to resume a training run. I think I've not seen this happen before but it is worth looking if this is reproducible. Are you able to reproduce this? Can you try to run a small test by resuming a run in a similar way after on 5-6 epochs to see if this issue still exists?

**Comment by wartek69** [comment]
@AyushExel I do not seem to be able to reproduce this weird issue either. However, I did notice that the previous trainings I did were using a K80 GPU and the latest training I did was a T4 GPU. I'm not sure this is relevant but I wanted to mention this.
Further I was wondering, since this bug messed up my checkpoints, can I revert back to an earlier checkpoint to continue training from the correct checkpoint, is it documented somewhere on how to do this?
